Rubberduck.Parsing/TypeResolvers/SetTypeResolver.cs

```diff
@@ -0,0 +1,268 @@
+﻿using System;
+using System.Linq;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.TypeResolvers
+{
+    public class SetTypeResolver :  ISetTypeResolver
+    {
+        public const string NotAnObject = "NotAnObject";
```

Comment:
Though it's very unlikely we'll encounter a string `NotAnObject` somewhere, one might want to consider adding certain characters which would be ordinary an illegal identifier name to make it even unlikely.

Reply:
from an earlier chat conversation around that I recall this string being as good as an illegal identifier, because the types are qualified with the project name and there is no built-in type with that name

Reply:
Exactly